lib/src/coap_message.dart

```diff
@@ -503,8 +503,12 @@ class CoapMessage {
   }
 
   /// URI path
+  // TODO: Apply proper percent-encoding
```

Comment:
The `convert` library has an encoder/decoder for percent as well (in case we need to consider other characters as well): https://github.com/dart-lang/convert/tree/master/lib/src/percent

Reply:
Thank you for the link! The problem is that there are very specific instructions for the percent encoding (see https://datatracker.ietf.org/doc/html/rfc7252#section-6.5) and it seems to me that there is no way to define exceptions for which characters are supposed to be encoded :/ (I originally used the `decodeComponent` method, but it also converted sub-delimiters like the `=` sign).